File: frontend/clean_locales.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const LOCALES_DIR = path.join(__dirname, 'src', 'locales');

function cleanString(str) {
  if (typeof str !== 'string') return str;
  // Remove like "[JA] " or " [JA]" or "[JA]"
  return str.replace(/\[[A-Z]{2,3}\]\s*/g, '').replace(/\s*\[[A-Z]{2,3}\]/g, '').trim();
}

function cleanObject(obj) {
  if (typeof obj === 'string') {
    return cleanString(obj);
  }
  if (Array.isArray(obj)) {
    return obj.map(item => cleanObject(item));
  }
  if (typeof obj === 'object' && obj !== null) {
    const cleaned = {};
    for (const key in obj) {
      cleaned[key] = cleanObject(obj[key]);
    }
    return cleaned;
  }
  return obj;
}

const files = fs.readdirSync(LOCALES_DIR);

files.forEach(file => {
  if (file.endsWith('.json')) {
    const filePath = path.join(LOCALES_DIR, file);
    const content = fs.readFileSync(filePath, 'utf8');
    try {
      const data = JSON.parse(content);
      const cleanedData = cleanObject(data);
      fs.writeFileSync(filePath, JSON.stringify(cleanedData, null, 2), 'utf8');
      console.log(`Cleaned ${file}`);
    } catch (e) {
      console.error(`Error processing ${file}: ${e.message}`);
    }
  }
});

File: frontend/clean_locales.py
import os
import json
import re

LOCALES_DIR = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'src', 'locales')

def clean_string(s):
    if not isinstance(s, str):
        return s
    s = re.sub(r'\[[A-Z]{2,3}\]\s*', '', s)
    s = re.sub(r'\s*\[[A-Z]{2,3}\]', '', s)
    return s.strip()

def clean_object(obj):
    if isinstance(obj, str):
        return clean_string(obj)
    elif isinstance(obj, list):
        return [clean_object(item) for item in obj]
    elif isinstance(obj, dict):
        return {k: clean_object(v) for k, v in obj.items()}
    return obj

def main():
    if not os.path.exists(LOCALES_DIR):
        print(f"Directory {LOCALES_DIR} does not exist.", flush=True)
        return
        
    for filename in os.listdir(LOCALES_DIR):
        if filename.endswith('.json'):
            filepath = os.path.join(LOCALES_DIR, filename)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    data = json.load(f)
                
                cleaned_data = clean_object(data)
                
                with open(filepath, 'w', encoding='utf-8') as f:
                    json.dump(cleaned_data, f, indent=2, ensure_ascii=False)
                print(f"Cleaned {filename}", flush=True)
            except Exception as e:
                print(f"Error processing {filename}: {e}", flush=True)

if __name__ == '__main__':
    main()
